### ISSUE TITLE
Alerts don't ignore tlsSkipVerify for Elasticsearch

### ISSUE THREAD

**Issue by mathewmeconry** [grafana/grafana#38067]
**What happened**:
Alerts are ignoring the `tlsSkipVerify` option for the Elasticsearch data source.
The query itself is working fine and returns the expected result:
![image](https://user-images.githubusercontent.com/4057473/130064782-6262b346-cbe4-4f0e-b996-0760a523d88c.png)

But the alerts are failing to query the data due to a TLS error:
`t=2021-08-19T11:55:01+0000 lvl=eror msg="Alert Rule Result Error" logger=alerting.evalContext ruleId=1 name="Redacted" error="request handler error: failed to query data: Post \"https://Redacted/_msearch?max_concurrent_shard_requests=256\": x509: certificate is valid for domain, not other-domain" changing state to=keep_state`

**What you expected to happen**:
The alerts query also ignores the certificate error as the query does.

**How to reproduce it (as minimally and precisely as possible)**:
- Create an Elasticsearch setup with SSL and a self-signed certificate.
- Setup the Grafana Datasource
  ```yaml
  apiVersion: 1
  datasources:
    - name: Elasticsearch Logs
      type: elasticsearch
      access: proxy
      url: https://elasticsearch:9200
      database: INDEXNAME
      basicAuth: true
      basicAuthUser: USERNAME
      isDefault: false
      editable: false
      secureJsonData:
        basicAuthPassword: PASSWORD
      jsonData:
        timeField: "@timestamp"
        esVersion: "7.10.0"
        logMessageField: log
        timeInterval: 10s
        tlsSkipVerify: true
        xpack: true  
  ```

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.0
- Data source type & version: Elasticsearch
- OS Grafana is installed on: Alpine Linux v3.13
- User OS & Browser: Not needed for this error
- Grafana plugins: Only Defaults
- Others: Installed with the helm chart from https://grafana.github.io/helm-charts (v6.15.0)
helm chart values:
```yaml
grafana:
  deploymentStrategy:
    type: Recreate
  extraSecretMounts:
    - name: auth-google-oauth-secret-mount
      secretName: auth-google-oauth-secret
      defaultMode: 0440
      mountPath: /etc/secrets/auth_google_oauth
      readOnly: true
  grafana.ini:
    server:
      domain: REDACTED
      root_url: "https://%(domain)s/"
    auth.google:
      enabled: true
      client_id: $__file{/etc/secrets/auth_google_oauth/client_id}
      client_secret: $__file{/etc/secrets/auth_google_oauth/client_secret}
      scopes: https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email
      auth_url: https://accounts.google.com/o/oauth2/auth
      token_url: https://accounts.google.com/o/oauth2/token
      allowed_domains: REDACTED
      allow_sign_up: true
  ingress:
    enabled: true
    path: /*
    pathType: ImplementationSpecific
    hosts:
      - REDACTED
    annotations:
      networking.gke.io/v1beta1.FrontendConfig: "grafana-ingress-config"
      networking.gke.io/managed-certificates: "grafana"
      kubernetes.io/ingress.class: "gce"
  
  persistence:
    enabled: true
  
  sidecar:
    datasources:
      enabled: true
    notifiers:
      enabled: true

  admin:
    existingSecret: "grafana-admin"

  imageRenderer:
    enabled: true
```

**Comment by mathewmeconry** [comment]
Tested it now with the new Grafana 8 alerts (by enabling the feature flag `ngalert`) and the same error happens:
`https://DOMAIN/_msearch?max_concurrent_shard_requests=5": x509: certificate is valid for OTHER-DOMAIN, not DOMAIN`

**Comment by mathewmeconry** [comment]
I switched now the url to match the url in the certificate. 
Now I get the error `t=2021-08-19T14:44:20+0000 lvl=eror msg="Alert Rule Result Error" logger=alerting.evalContext ruleId=2 name="CPU Alert" error="request handler error: failed to query data: Post \"https://DOMAIN:9200/_msearch?max_concurrent_shard_requests=256\": x509: certificate signed by unknown authority" changing state to=keep_state`

so queries for panels are still working without any issues. 
alerts are still ignoring the `tlsSkipVerify` flag

**Comment by redbaron4** [comment]
I have same issue not with elasticsearch (which appears to work fine) but graphite. The graphite server is behind an nginx proxy (HTTPS only) which uses a self signed certificate. Any "Test Rule" errors out with

```
"request handler error: failed to query data: Post "https://XXX/graphite/render": x509: certificate signed by unknown authority"
```

I have not enabled `ngalert`

**Comment by mathewmeconry** [comment]
Downgrading to 8.0.5 also resolved all of the issues mentioned above
thanks @redbaron4

**Comment by Elfo404** [comment]
Closing this as it should be fixed by #37797. Can you try with Grafana 8.1.2?

**Comment by mathewmeconry** [comment]
@Elfo404 version 8.1.2. resolves the issue
thanks a lot
